library/core/src/fmt/num.rs

```diff
@@ -248,6 +253,13 @@ macro_rules! impl_Display {
                 issue = "none"
             )]
             pub fn _fmt<'a>(self, buf: &'a mut [MaybeUninit::<u8>]) -> &'a str {
```

Comment:
This needs to be an unsafe function because it makes assumptions on the size of `buf`.